### PR TITLE
Add React Storybook + Stories for <Button />

### LINKED ISSUE #1080: Create Stories for One Component in React Storybook
We'd like to move forward with using React Storybook to create a component library! This is a follow-up to #1074

Acceptance Criteria:
1. Developer is able to use `npm run storybook` to boot up our library
2. The library should contain stories for one ui component

### PR DESCRIPTION
**Pull request by joshri**
Closes #1080 

Adds React Storybook plus stories for the Button component, as well as defaults and a `loading` prop to reduce Button-related code across the ui. Run storybook with `npm run storybook` to check it out!

BIG PROBLEM: Unfortunately, importing fonts into our theme with Parcel breaks Storybook - so fonts were removed to code out the initial implementation.

### CODE REVIEW

**Review by jpellizzari — COMMENTED** [review]
Screenshot from 2021-11-19 16-26-20

The primary button is looking a little...slim? :wink:

**Comment by jpellizzari** [comment]
@joshri Please incorporate the changes made here into this branch: https://github.com/weaveworks/weave-gitops/pull/1130

**Review by jpellizzari — COMMENTED** [review]
Questions around all of the `.bind` stuff

File: ui/pages/ApplicationAdd.tsx [hunk 1]
```diff
@@ -77,15 +77,11 @@ const SuccessMessage = styled(
...
-                        View Applications
-                      </Button>
+                      <Button>View Applications</Button>
                     </Link>
                   ) : (
                     <a target="_blank" href={link}>
-                      <Button color="primary" variant="outlined">
-                        View Open Pull requests
-                      </Button>
+                      <Button>View Open Pull requests</Button>
```
[the remark this answers is not in the document]
Comment (joshri) [reply ui/pages/ApplicationAdd.tsx:84]: I made the default Button `color='primary'` and `variant='outlined' ` - it already seemed like it was the default anyway?

File: ui/stories/Button.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      control: null,
+    },
+  },
+} as Meta;
+
+const Template: Story<Props> = (args) => <Button {...args}>Weaveworks</Button>;
+
+export const Default = Template.bind({});
+Default.args = { color: "primary", variant: "outlined" };
+// export const AuthButton = Template.bind({});
```
[the remark this answers is not in the document]
Comment (joshri) [reply ui/stories/Button.stories.tsx:56]: This is to remind me to add the 'auth-button' class to the story - the new auth buttons aren't in the branch

File: ui/stories/Alert.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      defaultValue: "success",
+    },
+  },
+} as Meta;
+
+const Template: Story<Props> = (args) => <Alert {...args} />;
+
+export const Success = Template.bind({});
+Success.args = { title: "Title", message: "Message / JSX" };
+export const Error = Template.bind({});
```
[the remark this answers is not in the document]
Comment (joshri) [reply ui/stories/Alert.stories.tsx:29]: I was following the syntax here: https://storybook.js.org/docs/react/writing-stories/args

The Template applies to every story - Success.args is what's passed down to the Success story. It saves you from typing out that Template for every story you create.

File: ui/stories/Button.stories.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jpellizzari) [reply ui/stories/Button.stories.tsx:56]: Let's not leave commented out stuff without more context. If we can add a comment for why this is commented, that would be nice.

File: ui/components/Alert.tsx [hunk 4]
```diff
@@ -8,17 +8,24 @@ import * as React from "react";
...
-  className?: string;
+/** Alert Properties */
+export interface Props {
+  /** string of one of the colors from our `MuiTheme` - also sets the corresponding material icon - see /ui/lib/theme.ts and https://mui.com/customization/theming/ */
+  severity?: AlertProps["severity"];
+  /** Overrides `justify-content: flex-start` (left) to render the Alert in the center of it's 100% width `<Flex />` component */
   center?: boolean;
+  /** text for Mui's `<AlertTitle />` component */
   title?: string;
+  /** Appears under `title` */
```
[the remark this answers is not in the document]
Comment (jpellizzari) [reply ui/components/Alert.tsx:19]: No just that we can do this with comments is cool

**Review by jpellizzari — APPROVED** [review]

**Pull request merged**